stores/menus.js

```diff
@@ -54,57 +81,69 @@ export const useMenusStore = defineStore('menus', {
         },
-        getContentTypeByName(name){
+        getContentTypeByName(name, locale){
             
             return Object.values(this.contentTypes).find((ct)=> { 
+                if(locale) return (ct.name.toLowerCase() === noCase(name)  || ct.plural.toLowerCase() === noCase(name)) && ct.langcode === locale;
                 return ct.name.toLowerCase() === noCase(name) || ct.plural.toLowerCase() === noCase(name)});
         },
         isContentTypeId(id){
             return (Object.values(this.contentTypes).map((ct)=> ct.drupalInternalId)).includes(Number(id));
         },
-        getContentType(name,country, locale){
-            const typeMapIds  = { news:2, event:3, 'learning-resource':4, project:5, 'basic-page':6, 'government-ministry-or-institute':8, ecosystem:9, 'protected-area':10, 'biodiversity-data':11, document:12, 'related-website':13, other:15, 'image-or-video':16 };
-            const hasKey      = this.contentTypes[name];
-            const id          = typeMapIds[name]
+        getContentType(name,locale){
+            const id          = typeMapIds[name];
             const contentType =  this.getContentTypeById(id, locale);
         
             return contentType;
         },
         getContentTypeData(name,country, locale){
-            const typeMapIds  = { news:2, event:3, 'learning-resource':4, project:5, 'basic-page':6, 'government-ministry-or-institute':8, ecosystem:9, 'protected-area':10, 'biodiversity-data':11, document:12, 'related-website':13, other:15, 'image-or-video':16 };
-            const hasKey      = this.contentTypes[name];
-            const id          = typeMapIds[name]
+            const id          = typeMapIds[name];
             const contentType =  this.getContentTypeById(id, locale);
         
             return country? contentType?.dataMap[country] : contentType.data;
         },
         isInMainMenuByContentTypeId(id){
-            if(!id) return false;
+            if(!id || !this?.main?.length) return false;
+    
             for(let i = 0; i < this.main.length; i++)
                 if(this.isInMenuByContentTypeId(this.main[i], id)) return this.isInMenuByContentTypeId(this.main[i], id)
         
             return false;
         },
+        isInContentTypesById(id){
+            if(!id) return false;
+            if(!this?.main?.length) return false;
+
+            for(let i = 0; i < this.main.length; i++)
+                if(this.main[i].children?.length)
+                    for(let j = 0; j < this.main[i].children.length; j++)
+                        if(this.main[i].children[j].length)
+                            if(this.isInMenuByContentTypeId(this.main[i].children[j], id)) 
+                                return this.isInMenuByContentTypeId(this.main[i].children[j], id)
```

Comment:
why not use `for... of`
More or less
```suggestion
for(let entry of this.main.length || [])
    for(let child of entry.children || [])
        if(this.isInMenuByContentTypeId(child, id)) 
            return this.isInMenuByContentTypeId(child, id)
```

Reply:
applied thanks